packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -18,126 +18,127 @@ import H from '../constants';
 const watchmanURL =
   'https://facebook.github.io/watchman/docs/troubleshooting.html';
 
-function isDescendant(root: string, child: string): boolean {
-  return child.startsWith(root);
+class WatchmanError extends Error {
+  constructor(error) {
+    super(
+      `Watchman error: ${error.message.trim()}. Make sure watchman ` +
+        `is running for this project. See ${watchmanURL}.`,
+    );
+    const stack = error.stack;
```

Comment:
I guess I can merge stack traces instead since `error.stack` should never be empty.